src/iperf_api.c

```diff
@@ -1273,12 +1276,15 @@ iperf_parse_arguments(struct iperf_test *test, int argc, char **argv)
         !(server_rsa_private_key && test->server_authorized_users)) {
          i_errno = IESETSERVERAUTH;
         return -1;
-    } else if (test->role == 's' && server_rsa_private_key && test_load_private_key_from_file(server_rsa_private_key) < 0){
-        i_errno = IESETSERVERAUTH;
-        return -1;
-    } else {
+    } else if (test->role == 's' && server_rsa_private_key == NULL) {
```

Comment:
The check for `server_rsa_private_key` got inverted here. The net effect is that authentication succeeds even if the wrong password is given.